src/components/MultiHopTrade/components/LimitOrder/components/CancelLimtOrder.tsx

```diff
@@ -0,0 +1,222 @@
+import { useActions } from 'hooks/useActions'
+import { useErrorHandler } from 'hooks/useErrorToast/useErrorToast'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { useCancelLimitOrdersMutation } from 'state/apis/limit-orders/limitOrderApi'
+import { limitOrderSlice } from 'state/slices/limitOrderSlice/limitOrderSlice'
+import { selectOrderToCancel } from 'state/slices/limitOrderSlice/selectors'
+import { selectAssetById, selectFeeAssetById } from 'state/slices/selectors'
+import { useAppSelector, useSelectorWithArgs } from 'state/store'
+
+import { SwapperIcon } from '../../TradeInput/components/SwapperIcon/SwapperIcon'
+
+const cardBorderRadius = { base: '2xl' }
+
+export const CancelLimitOrder = () => {
+  const wallet = useWallet().state.wallet
+  const { setOrderToCancel } = useActions(limitOrderSlice.actions)
+  const { showErrorToast } = useErrorHandler()
+
+  const orderToCancel = useAppSelector(selectOrderToCancel)
+
+  const [cancelLimitOrders, { data: wasCancellationSuccessful, error, isLoading, reset }] =
+    useCancelLimitOrdersMutation()
+
+  useEffect(() => {
+    if (!error) return
+
+    const description = (error as CowSwapError).description ?? 'Unknown Error'
+
+    // TODO: Actually render a translated error description.
+    showErrorToast(description)
+  }, [error, showErrorToast])
+
+  useEffect(() => {
+    // Must explicitly check for false, because undefined means a cancellation isn't completed.
+    if (wasCancellationSuccessful !== false) return
+
+    showErrorToast(
+      `Failed to cancel order uid ${orderToCancel?.order.uid}`,
+      'limitOrder.cancel.cancellationFailed',
+    )
+  }, [error, orderToCancel?.order.uid, showErrorToast, wasCancellationSuccessful])
+
+  const handleClose = useCallback(() => {
+    reset()
+    setOrderToCancel(undefined)
+  }, [setOrderToCancel, reset])
+
+  const handleRequestCancellation = useCallback(async () => {
+    if (!orderToCancel || !wallet) {
+      return
+    }
+
+    await cancelLimitOrders({ wallet, ...orderToCancel })
+    setOrderToCancel(undefined)
+  }, [orderToCancel, wallet, cancelLimitOrders, setOrderToCancel])
+
+  const sellAsset = useSelectorWithArgs(selectAssetById, orderToCancel?.sellAssetId ?? '')
+  const buyAsset = useSelectorWithArgs(selectAssetById, orderToCancel?.buyAssetId ?? '')
+  const feeAsset = useSelectorWithArgs(selectFeeAssetById, orderToCancel?.sellAssetId ?? '')
+
+  const limitPrice = useMemo(() => {
+    if (!orderToCancel || !sellAsset || !buyAsset) return
+    const buyAmountCryptoPrecision = fromBaseUnit(orderToCancel.order.buyAmount, buyAsset.precision)
+    const sellAmountCryptoPrecision = fromBaseUnit(
+      orderToCancel.order.sellAmount,
+      sellAsset.precision,
+    )
+    return bn(buyAmountCryptoPrecision).div(sellAmountCryptoPrecision).toFixed()
+  }, [buyAsset, orderToCancel, sellAsset])
+
+  const expiryText = useMemo(() => {
+    const validTo = orderToCancel?.order.validTo
```

Comment:
nitpick: early return on `!otderToCancel` instead, `validTo` is always defined assuming `orderToCancel` is (which it should and we can ensure with an early return)

Reply:
2151427d82837a2c91febcaed514dcce0fdc662d